From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 9 Jul 2010 13:35:14 +0200
Subject: [PATCH] add linker option to mklib call

When cross compiling, we have to handle the (cross-)linker to the mklib
call.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/gallium/winsys/drm/i965/xorg/Makefile  |    2 +-
 src/gallium/winsys/drm/intel/xorg/Makefile |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gallium/winsys/drm/i965/xorg/Makefile b/src/gallium/winsys/drm/i965/xorg/Makefile
index c25726b..3e00035 100644
--- a/src/gallium/winsys/drm/i965/xorg/Makefile
+++ b/src/gallium/winsys/drm/i965/xorg/Makefile
@@ -35,7 +35,7 @@ TARGET_STAGING = $(TOP)/$(LIB_DIR)/gallium/$(TARGET)
 all default: $(TARGET) $(TARGET_STAGING)
 
 $(TARGET): $(OBJECTS) Makefile $(GALLIUMDIR)/state_trackers/xorg/libxorgtracker.a $(LIBS)
-	$(TOP)/bin/mklib -noprefix -o $@ \
+	$(TOP)/bin/mklib -noprefix -linker '$(CC)' -o $@ \
 	$(OBJECTS) $(LIBS) $(shell pkg-config --libs libdrm) -ldrm_intel
 
 $(TOP)/$(LIB_DIR)/gallium:
diff --git a/src/gallium/winsys/drm/intel/xorg/Makefile b/src/gallium/winsys/drm/intel/xorg/Makefile
index 14c2462..ad44715 100644
--- a/src/gallium/winsys/drm/intel/xorg/Makefile
+++ b/src/gallium/winsys/drm/intel/xorg/Makefile
@@ -34,7 +34,7 @@ DRIVER_DEFINES = \
 all default: $(TARGET)
 
 $(TARGET): $(OBJECTS) Makefile $(TOP)/src/gallium/state_trackers/xorg/libxorgtracker.a $(LIBS)
-	$(TOP)/bin/mklib -noprefix -o $@ \
+	$(TOP)/bin/mklib -noprefix -linker '$(CC)' -o $@ \
 	$(OBJECTS) $(LIBS) $(shell pkg-config --libs libdrm) -ldrm_intel
 
 clean:
-- 
1.7.1

